cocos/base/CCEventDispatcher.cpp

```diff
@@ -1190,7 +1190,13 @@ void EventDispatcher::sortEventListenersOfSceneGraphPriority(const EventListener
     _nodePriorityIndex = 0;
     _nodePriorityMap.clear();
 
-    visitTarget(rootNode, true);
+    Node *notificationNode = Director::getInstance()->getNotificationNode();
+    if(notificationNode) {
+        visitTarget(notificationNode, true);
+        visitTarget(rootNode, false);
```

Comment:
This fix is all about event handling. Logically notification node is always "on top" of scene, so it should receive events (e.g. touches) first. 

If you use CCMenu on both notification node and scene without this fix there is no guarantee that notification node's one will be triggered (if both are visually on the same location).